### DISCUSSION TITLE
Serialize FileSystemHandle (and directory handle)

### DISCUSSION THREAD

**Discussion by bpafoshizle** [vuejs/pinia#2271] [Help and Questions] [answered]
Hello,

I’m trying to see if it’s possible to serialize a FileSystemHandle or FileSystemDirectoryHandle object to local storage. I’m building an app that lets a user pick a local directory and file through a   file picker which returns a FileSystemHandle. This has to be from a user initiated action (like a button click) and also have permission to access the file system granted. According to [this](https://developer.chrome.com/articles/file-system-access/), the handles are serializable (although the user-granted permission is not and has to be regranted between sessions) and they show a method of serializing the handle objects to IndexedDB. I know by default Pinia is using Local Storage for persistence, and inspecting this, the handles never get properly persisted (there are only empty objects), but the handles are in the in-memory store and can pass between components just fine. 

My question is whether it’s something to do with local storage vs IndexedDB or if Pinia is not serializing them properly for some reason. If someone can point me in the right direction I could even attempt a PR to fix (if applicable).

**Comment by posva** [comment, marked as answer]
Pinia doesn’t persist anything by default so you are free to use or created your own custom storage that uses indexdb

**Reply by bpafoshizle** [reply]
I realized after asking this question I’m using a plug-in for this [pinia-plugin-persistedstate](https://github.com/prazdevs/pinia-plugin-persistedstate/tree/main), not Pinia directly, so I should have asked there. My apologies. I stopped working on my project for a while and just came back to it, so I’d forgotten that nuance. Thanks for the reply!
